### ISSUE TITLE
Dependency Injection validation issue with IHttpMessageHandlerBuilderFilter in Sentry 3.0.2

### ISSUE THREAD

**Issue by Turnerj** [getsentry/sentry-dotnet#785]
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io)

Which SDK and version?
.NET 3.0.2 (running on .NET 5)

### Steps to Reproduce

1. Create a new/blank ASP.NET Core project for .NET 5.
2. In `Program.cs`, have the following:

```csharp
public class Program
{
	public static void Main(string[] args)
	{
		CreateHostBuilder(args).Build().Run();
	}

	public static IHostBuilder CreateHostBuilder(string[] args) =>
		Host.CreateDefaultBuilder(args)
			.ConfigureWebHostDefaults(webBuilder =>
			{
				webBuilder
					.UseStartup<Startup>()
					.UseSentry();
			});
}
```
3. In `Startup.cs` have the following:

```csharp
public class Startup
{
	public void ConfigureServices(IServiceCollection services)
	{
		services.AddHttpClient(); // This line causes conflict with Sentry 3.0.2 (works fine in 3.0.1)
	}

	public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
	{
		if (env.IsDevelopment())
		{
			app.UseDeveloperExceptionPage();
		}

		app.UseRouting();

		app.UseEndpoints(endpoints =>
		{
			endpoints.MapGet("/", async context =>
			{
				await context.Response.WriteAsync("Hello World!");
			});
		});
	}
}
```

4. Try to debug the application

### Expected Result

It would work? Everything about the patch seemed like it would just run fine. 🤷‍♂️

### Actual Result

An exception is thrown during startup when debugging as Microsoft Dependency Injection has validation enabled by default during development/debugging.

> System.AggregateException: 'Some services are not able to be constructed (Error while validating the service descriptor 'ServiceType: Microsoft.Extensions.Http.DefaultHttpClientFactory Lifetime: Singleton ImplementationType: Microsoft.Extensions.Http.DefaultHttpClientFactory': Cannot consume scoped service 'System.Collections.Generic.IEnumerable`1[Microsoft.Extensions.Http.IHttpMessageHandlerBuilderFilter]' from singleton 'Microsoft.Extensions.Http.DefaultHttpClientFactory'.)'

### Notes

The issue seems to relate to changes in #784, specifically this line: https://github.com/getsentry/sentry-dotnet/pull/784/files#diff-7d117571d3f4e39dc910da7ee00ab9691f575b9e3064a02cb9d1f689cae5d188R37-R40

At this stage, I haven't been able to isolate the issue any further than this. It seems to be conflicting with some use of injecting a `HttpClient` or `IHttpClientFactory` somewhere in my own `ConfigureServices` method in my startup class.

I'll post more details when I have them!

**EDIT:** The issue seems to stem from any injection of a `HttpClient` and/or calling `AddHttpClient` myself in my code. Going to look to see if I can write a test for the repo that can trigger this issue too.

**Comment by Turnerj** [comment]
In the [existing test](https://github.com/getsentry/sentry-dotnet/blob/2827a1bfd74c1c7619aacf13d5893644346f0636/test/Sentry.AspNetCore.Tests/SentryHttpMessageHandlerBuilderFilterTests.cs#L38-L90), if you modify it to enable dependency validation, it does trigger the error that I receive.

```diff
var server = new TestServer(new WebHostBuilder()
    .UseSentry()
+    .UseDefaultServiceProvider(opts =>
+    {
+        opts.ValidateOnBuild = true;
+        opts.ValidateScopes = true;
+    })
    .ConfigureServices(services =>
    {
```

Now even with this in mind, I'm not entirely sure who to "blame" here - is this Sentry using it right but bad docs from MS about `AddHttpClient` etc? Is this a bug in `Microsoft.DependencyInjection`? I don't know...

**Comment by Turnerj** [comment]
As an additional aside - I haven't yet encountered an issue with a `HttpClient` etc with the validation disabled though I don't exactly want to disable DI validation for development as it is extremely useful. Hopefully we can get to the bottom of the issue. 🙂

**Comment by rogerfar** [comment]
Can confirm this issue, downgraded to 3.0.1 and works again.

**Comment by Tyrrrz** [comment]
Thanks for reporting, it should be an easy to bug to fix.
A good idea for us in the future: enable container validation in tests.

**Comment by Turnerj** [comment]
Hey @Tyrrrz - thanks for getting this sorted so fast!

**Comment by Tyrrrz** [comment]
@Turnerj no problem.
The fix should be included in version 3.0.3 deployed today.

https://github.com/getsentry/sentry-dotnet/releases/tag/3.0.3
